### PR TITLE
feat(flows): prefill copilot input from Suggested Workflows instead of inline build

### PR DESCRIPTION
**Pull request by graycyrus**
## Summary

- The Suggested Workflows "Build this" button no longer runs an inline `workflow_builder` chat turn and renders a `WorkflowProposalCard` in the suggestions list — it now creates a new blank flow (named from the suggestion, mirroring `WorkflowPromptBar`'s instant-create path) and navigates to its canvas.
- The suggestion's `build_prompt` is **pre-filled into the copilot's composer** on the canvas — never auto-sent. The user reviews/edits the brief and presses Send themselves.
- The suggestion is marked `built` (and dropped from the active list) once the flow is created, matching the old "Save & enable" behavior, so Flow Scout doesn't immediately re-suggest it.
- The existing instant-create (`copilotBuild`, auto-sends) and "Fix with agent" (`copilotRepair`) seed paths are untouched — this adds a third, sibling `copilotPrefill` seed that only populates text.

## Problem

Previously, clicking "Build this" silently kicked off an agent turn against the suggestion's raw `build_prompt` with no chance for the user to review or edit the brief before the agent started building — and the resulting proposal rendered inline in a list of suggestion cards rather than in the canvas where the rest of the copilot workflow lives.

## Solution

- `app/src/components/flows/SuggestedWorkflows.tsx` — `onBuild` now calls `createFlow(name, createBlankWorkflowGraph(...), true)` (same `require_approval: true` default as the prompt bar) and `navigate('/flows/:id', { state: { copilotPrefill: { text: suggestion.build_prompt } } })`. Marks the suggestion built via `markSuggestionBuilt` at navigate time. Removes the `useWorkflowBuilderChat` dependency and the inline `WorkflowProposalCard` rendering entirely. The button shows "Opening…" (was "Building…") while the flow is being created.
- `app/src/pages/FlowCanvasPage.tsx` — adds a `CopilotPrefillSeed` type + `asCopilotPrefillSeed` narrower (mirrors the existing `CopilotBuildSeed`/`asCopilotBuildSeed`), threads it through `FlowEditor` as `prefillSeed`/`onPrefillSeedConsumed`, and strips only `copilotPrefill` from `location.state` once consumed (same pattern as `clearBuildSeed`, preserving sibling seeds like `copilotRepair`).
- `app/src/components/flows/WorkflowCopilotPanel.tsx` — adds a `prefillSeed` prop with a once-per-mount effect (`prefillSentRef`) that calls `setText(...)` and focuses the composer — it deliberately never calls `send()`. Reports consumption synchronously (no async turn to await, unlike `buildSeed`/`repairSeed`) so the host can strip the route seed immediately.
- i18n: renamed `flows.suggest.building` → `flows.suggest.opening` ("Opening…") across `en` + all 13 locales, since the button's busy state now reflects "opening the canvas" rather than "the agent is building".

## Submission Checklist

- [x] Tests added or updated (happy path + at least one failure / edge case) — rewrote `SuggestedWorkflows.test.tsx`'s "Build this" coverage for create+navigate+prefill (including a `createFlow` failure path), added `WorkflowCopilotPanel.test.tsx` coverage for `prefillSeed` (populates without sending, survives rerenders without clobbering user edits, doesn't re-fire on remount), and added `FlowCanvasPage.test.tsx` coverage for `asCopilotPrefillSeed` + route-state wiring/clearing.
- [x] **Diff coverage ≥ 80%** — ran the affected Vitest suites locally (`SuggestedWorkflows`, `WorkflowCopilotPanel`, `FlowCanvasPage`, `WorkflowPromptBar`): 55/55 passing, all touched logic branches exercised. No Rust changed (`pnpm test:rust` N/A).
- [x] N/A: Coverage matrix — no existing Workflows/Flows feature ID in `docs/TEST-COVERAGE-MATRIX.md` (the tinyflows integration predates this matrix's coverage); this is a UX refinement of an already-shipped, uncatalogued surface, so no rows are added/removed/renamed.
- [x] N/A: no matrix rows changed (see above), so no feature IDs to list under `## Related`.
- [x] No new external network dependencies introduced — reuses the existing `flows_create`/`flows_discover`/`flows_dismiss_suggestion`/`flows_mark_suggestion_built` RPCs against the mock backend.
- [x] N/A: Manual smoke checklist — `docs/RELEASE-MANUAL-SMOKE.md` has no Workflows/Flows entries to update.
- [x] N/A: no linked GitHub issue for this change (user-directed UX refinement).

## Impact

- Runtime/platform: desktop UI only (`app/src`), no Rust/core changes.
- Behavior change is scoped to the Suggested Workflows "Build this" affordance; instant-create (prompt bar) and "Fix with agent" (failed-run repair) copilot seed paths are unaffected — verified via existing regression tests for both.

## Related

- Closes: N/A
- Follow-up PR(s)/TODOs: none

---

### Linear Issue

- Key: N/A
- URL: N/A

### Commit & Branch

- Branch: `feat/flows-build-prefills-copilot-input`
- Commit SHA: `d200638415387e4082ab293c6664ed55593a8181`

### Validation Run

- [x] `pnpm --filter openhuman-app format:check` — ran `pnpm format` (prettier + cargo fmt); no diffs beyond the intended files.
- [x] `pnpm typecheck` — clean.
- [x] Focused tests: `vitest run --config test/vitest.config.ts src/components/flows/SuggestedWorkflows.test.tsx src/components/flows/WorkflowCopilotPanel.test.tsx src/pages/__tests__/FlowCanvasPage.test.tsx src/components/flows/WorkflowPromptBar.test.tsx` — 55/55 passing.
- [x] Rust fmt/check (if changed): N/A — no Rust files touched.
- [x] Tauri fmt/check (if changed): N/A — no Tauri shell files touched.

### Validation Blocked

- `command:` N/A
- `error:` N/A
- `impact:` N/A

### Behavior Changes

- Intended behavior change: "Build this" on a Suggested Workflows card creates a new flow and opens its canvas with the copilot input pre-filled from `build_prompt`, instead of running an inline agent turn.
- User-visible effect: the user now lands in the canvas and must press Send themselves to kick off the build — they can review/edit the brief first. The suggestion still gets marked built/removed from the list at that point.

### Parity Contract

- Legacy behavior preserved: `WorkflowPromptBar`'s instant-create (`copilotBuild`, auto-sends) and `FlowRunsSidebar`'s "Fix with agent" (`copilotRepair`, auto-sends) paths are unchanged — both have passing regression coverage confirming no cross-talk with the new `copilotPrefill` seed.
- Guard/fallback/dispatch parity checks: the new `prefillSeed` effect uses the same once-per-mount ref-guard pattern (`prefillSentRef`) as `buildSentRef`/`repairSentRef`, and the host clears the ephemeral route seed the same way `clearBuildSeed` does (deletes only its own key, preserving siblings) — covered by dedicated tests.

### Duplicate / Superseded PR Handling

- Duplicate PR(s): none known.
- Canonical PR: this one.
- Resolution (closed/superseded/updated): N/A

## Summary by CodeRabbit

* **New Features**
  * Suggested workflows now create and open a blank workflow directly in the canvas, prefilled for Copilot.
  * Copilot opens automatically when a suggested workflow is selected, with the first message using the correct build mode.
  * While a workflow is being created, “Build this” buttons are temporarily disabled and the card shows an “Opening…” state.
* **Bug Fixes**
  * Improved handling of workflow creation failures: no navigation occurs and the suggestion remains available.
  * Copilot prefilled text is applied once and won’t reappear after being consumed.
* **Localization**
  * Updated “suggested workflow” loading copy across supported languages from “Building…” to “Opening…”.

### CODE REVIEW

File: app/src/components/flows/SuggestedWorkflows.tsx [hunk 1]
```diff
@@ -148,16 +153,42 @@ export default function SuggestedWorkflows() {
...
+        // Slack/Gmail/HTTP/code nodes cannot fire unattended, matching
+        // `WorkflowPromptBar`'s instant-create default.
+        const flow = await createFlow(
+          name,
+          createBlankWorkflowGraph(name, t('flows.nodeKind.trigger')),
+          true
+        );
+        log('onBuild: created id=%s — opening canvas with prefill seed', flow.id);
+        removeSuggestion(suggestion.id);
+        void markSuggestionBuilt(suggestion.id).catch(e =>
```
[the remark this answers is not in the document]
Comment (graycyrus) [reply app/src/components/flows/SuggestedWorkflows.tsx]: Fixed in a3248498a: removed the persistent `markSuggestionBuilt(suggestion.id)` call from `onBuild` entirely.

Decision on the UX: I opted for the safer choice you outlined — leave the suggestion un-built server-side rather than build a new save-hook. This path only creates a blank flow + an unsent prefill; there's no clean signal today from the canvas's Save (which lives on a totally separate `/flows/:id` route/component) back to the originating suggestion id, and wiring one up would mean threading a `suggestionId` through `copilotPrefill` -> `FlowCanvasPage` -> `handleSave`, plus reasoning about idempotency/renames/multiple saves — more surface than this fix should carry. So: no persistent 'built' mark until there's a real save-aware hook; the suggestion can simply resurface on the next Flow Scout discovery run if abandoned. I kept the existing optimistic local removal (`removeSuggestion`) so the card still drops from the CURRENT session's list right after the user acts on it once, just not persisted as 'built'.

Updated `SuggestedWorkflows.test.tsx`'s test to assert `markSuggestionBuilt` is never called on Build this, while the local list drop still happens.

File: app/src/components/flows/SuggestedWorkflows.tsx [hunk 2]
```diff
@@ -148,16 +153,42 @@ export default function SuggestedWorkflows() {
...
+          log('markSuggestionBuilt failed: %o', e)
+        );
+        navigate(`/flows/${flow.id}`, {
+          state: { copilotPrefill: { text: suggestion.build_prompt } },
```
[the remark this answers is not in the document]
Comment (graycyrus) [reply app/src/components/flows/SuggestedWorkflows.tsx]: Fixed in a3248498a. Verified both reference points:
- Old inline path: `request: { mode: 'create', instruction: suggestion.build_prompt }` — no `flow_id`, because back then nothing was persisted until the proposal card's Save.
- Prompt bar's `copilotBuild`: `buildSeed` auto-sends `mode: 'build'` (with `flowId`) once the blank flow already exists — Rust's `BuildMode::Build` doc/brief literally says 'the flow already exists (created blank just now) — design the graph and verify it with dry_run_workflow'.

Since this new path *also* creates the blank flow before navigating (that's the whole point of the refactor), the correct mode for the first Send is `build`, matching the prompt bar's precondition exactly — not `create`, which was only right for the old architecture where no flow existed yet.

Implementation: threaded `mode` through the seed as you suggested —
- `SuggestedWorkflows.tsx`: `state: { copilotPrefill: { text: suggestion.build_prompt, mode: 'build' } }`
- `FlowCanvasPage.tsx`: `CopilotPrefillSeed.mode?: 'build' | 'create'`, parsed by `asCopilotPrefillSeed` (defaults to `'build'` if omitted/invalid, so an older/partial route state still gets the right mode)
- `WorkflowCopilotPanel.tsx`: `prefillSeed.mode` is captured into a `pendingPrefillModeRef` when the seed populates the input; `submit()`'s FIRST call consumes it (uses `{ mode: prefillMode, instruction, graph, flowId }` instead of `revise`) and clears it, so every subsequent Send on the same mount falls back to the panel's normal `revise`. Falls back to `revise` defensively if `flowId` is somehow missing (mirrors `buildSeed`'s own fallback).

Added `WorkflowCopilotPanel.test.tsx` coverage: first Send after a `mode: 'build'` prefill dispatches `mode: 'build'`, the second Send on the same mount falls back to `revise`, an omitted seed mode defaults to `build`, and a missing `flowId` falls back to `revise`. Also extended `FlowCanvasPage.test.tsx`'s `asCopilotPrefillSeed` tests for the mode parsing/defaulting.

File: app/src/components/flows/SuggestedWorkflows.tsx [hunk 3]
```diff
@@ -51,12 +56,13 @@ function triggerLabelKey(hint?: string | null): string | null {
...
   suggestion: FlowSuggestion;
-  building: boolean;
+  /** True while this suggestion's blank flow is being created + navigated to. */
+  opening: boolean;
   onBuild: () => void;
   onDismiss: () => void;
 }
 
-function SuggestionCard({ suggestion, building, onBuild, onDismiss }: SuggestionCardProps) {
+function SuggestionCard({ suggestion, opening, onBuild, onDismiss }: SuggestionCardProps) {
```
[the remark this answers is not in the document]
Comment (graycyrus) [reply app/src/components/flows/SuggestedWorkflows.tsx]: Fixed in a3248498a: added a `buildInProgress` prop to `SuggestionCard`, computed as `openingId !== null` at the map call site, and used it for the button's `disabled` instead of the per-card `opening` flag (which stayed but is now only used for the 'Opening…' label text). So every card's "Build this" is disabled while any one build is in flight, not just the active card's.

Added a test asserting both the clicked card's button AND a second, untouched card's button are disabled once one build is in flight.
